From 16cc79f975816c0741711560be48fc498d4b4794 Mon Sep 17 00:00:00 2001
From: Kenneth Graunke <kenneth@whitecape.org>
Date: Sun, 19 Feb 2012 05:29:29 +0000
Subject: i965: Only set Last Render Target Select on the last FB write.

Fixes GPU hangs in OilRush, Trine, and Amnesia: The Dark Descent,
which all use MRT (multiple render targets).

Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=38720
Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=40059
Bugzilla: https://bugs.freedesktop.org/show_bug.cgi?id=45216
Reviewed-by: Eric Anholt <eric@anholt.net>
Signed-off-by: Kenneth Graunke <kenneth@whitecape.org>
(cherry picked from commit 172bb92db1a3c317867d9cfec6f15c09c37a0f6c)

pixel scoreboard comment changed to make it apply to 7.11 branch.
Signed-off-by: Robert Hooker <sarvatt@ubuntu.com>
---

diff --git a/src/mesa/drivers/dri/i965/brw_eu_emit.c b/src/mesa/drivers/dri/i965/brw_eu_emit.c
index e7370f3..6a2a42a 100644
--- a/src/mesa/drivers/dri/i965/brw_eu_emit.c
+++ b/src/mesa/drivers/dri/i965/brw_eu_emit.c
@@ -2086,7 +2086,7 @@ void brw_fb_WRITE(struct brw_compile *p,
 			    msg_type,
 			    msg_length,
 			    header_present,
-			    1,	/* pixel scoreboard */
+			    eot,	/* pixel scoreboard */
 			    response_length,
 			    eot,
 			    0 /* send_commit_msg */);
